test/test_datasets.py

```diff
@@ -1790,5 +1790,77 @@ def test_targets(self):
                 assert item[6] == i // 3
 
 
+class LFWPeopleTestCase(datasets_utils.DatasetTestCase):
+    DATASET_CLASS = datasets.LFWPeople
+    FEATURE_TYPES = (PIL.Image.Image, int)
+    DEFAULT_CONFIG = dict(image_set="deepfunneled")
+    ADDITIONAL_CONFIGS = datasets_utils.combinations_grid(
+        train=(True, False),
+        image_set=('original', 'funneled', 'deepfunneled')
+    )
+    _IMAGES_DIR = {
+        "original": "lfw",
+        "funneled": "lfw_funneled",
+        "deepfunneled": "lfw-deepfunneled"
+    }
+
+    def inject_fake_data(self, tmpdir, config):
+        tmpdir = pathlib.Path(tmpdir) / "lfw-py"
+        os.makedirs(tmpdir, exist_ok=True)
+        split = "Train" if config["train"] else "Test"
+        return dict(
+            num_examples=self._create_images_dir(tmpdir, self._IMAGES_DIR[config["image_set"]], split),
+            split=split
+        )
+
+    def _create_images_dir(self, root, idir, split):
+        idir = os.path.join(root, idir)
+        os.makedirs(idir, exist_ok=True)
+        num_people = 4
+        num_examples = 0
+        flines = [str(num_people)]
+        for i in range(num_people):
+            name = datasets_utils.create_random_string(random.randint(5, 7)) + '_' \
+                + datasets_utils.create_random_string(random.randint(4, 7))
+            no = random.randint(1, 10)
+            flines.append(f"\n{name}\t{no}")
+            datasets_utils.create_image_folder(idir, name, lambda n: f"{name}_{n+1:04d}.jpg", no, 250)
+            num_examples += no
+        with open(pathlib.Path(root) / f"peopleDev{split}.txt", "w") as f:
+            f.writelines(flines)
+
+        return num_examples
+
+
+class LFWPairsTestCase(LFWPeopleTestCase):
+    DATASET_CLASS = datasets.LFWPairs
+    FEATURE_TYPES = ((PIL.Image.Image, PIL.Image.Image), int)
+
+    def _create_images_dir(self, root, idir, split):
+        idir = os.path.join(root, idir)
+        os.makedirs(idir, exist_ok=True)
+        num_pairs = 7  # effectively 7*2 = 14
+        with open(pathlib.Path(root) / f"pairsDev{split}.txt", "w") as file:
+            file.write(str(num_pairs))
+            for i in range(num_pairs):
+                name1 = datasets_utils.create_random_string(random.randint(5, 7)) + '_' \
+                    + datasets_utils.create_random_string(random.randint(4, 7))
```

Comment:
By putting this in a method, e.g. `_create_image_id`, we could avoid some duplication. These lines appear four times in total for both test cases.

Reply:
Yeah! That'd be great. Thanks for pointing it out.